django/applications/catmaid/control/graph2.py

```diff
@@ -312,7 +312,9 @@ def populate_connectors(chunkIDs, chunks, cs, connectors) -> None:
 
 
 def subgraphs(digraph, skeleton_id) -> Tuple[List, Tuple]:
-    chunks = list(digraph.subgraph(c).copy() for c in weakly_connected_components(digraph))
+    # For consistency and easier testing, sort chunks by size
```

Comment:
That's true and a good suggestion, thanks! I suppose you mean the lines after your commen:
```
 chunks = list(digraph.subgraph(c).copy() for c in \
            sorted(weakly_connected_components(digraph), key=len))
```
And I suppose you mean rewriting this like that:
```
 chunks = [digraph.subgraph(c).copy() for c in \
            sorted(weakly_connected_components(digraph), key=len)]
```
Or did you have something else in mind?

Reply:
I guess doing the following might be even better, because it avoids the immediate list from `sorted` and uses list comprehension too (I believe the `[]` are optional in this case):
```
chunks = sorted(digraph.subgraph(c).copy() for c in \
           weakly_connected_components(digraph), key=len))
```